app/scripts/loadout/dimLoadout.directive.js

```diff
@@ -33,8 +33,8 @@
         '        </form>',
         '      </div>',
         '      <span id="loadout-contents">',
-        '        <span ng-repeat="value in vm.types" class="loadout-{{ value }}">',
-        '          <div ng-repeat="item in vm.loadout.items[value]" ng-click="vm.equip(item)" id="loadout-item-{{:: $id }}" class="item" ng-class="{ \'complete\': item.complete}">',
+        '        <span ng-repeat="value in vm.types track by value" class="loadout-{{ value }}">',
+        '          <div ng-repeat="item in vm.loadout.items[value] track by item.id" ng-click="vm.equip(item)" id="loadout-item-{{:: $id }}" class="item" ng-class="{ \'complete\': item.complete}">',
```

Comment:
item.index is unique.  Items have a unique ID if they are not stackable.  Stackables have an id of 0.